scipy/optimize/_linprog.py

```diff
@@ -75,6 +75,12 @@ def linprog_verbose_callback(res):
         message : str
             A string descriptor of the exit status of the optimization.
     """
+    warn(
+        "linprog_verbose_callback is set to be depreciated in version 1.6. "
```

Comment:
If it is going to stay, better leave the version number ambiguous in case we cannot deprecate it for whatever reason e.g., `in the future versions`. `warn` needs a `stacklevel` argument.

Reply:
@mdhaber - thanks for the pick-up, I did a fair bit of financial/actuarial maths so "depreciated" is still part of my vocabulary.

@ilayn - would it be best to just cut it out? e.g. -> *linprog_verbose_callback is set to be ~depreciated~ deprecated.*?